src/ts_catalog/continuous_agg.c

```diff
@@ -848,6 +848,8 @@ ContinuousAgg *
 ts_continuous_agg_find_by_mat_hypertable_id(int32 mat_hypertable_id)
 {
 	ContinuousAgg *ca = NULL;
+	if (mat_hypertable_id == INVALID_HYPERTABLE_ID)
+		return ca;
```

Comment:
Isn't this something that the caller should be responsible for?

Reply:
It could be checked there of course - but I think there is no hard precondition to supply a valid argument.
The api doc doesn't explain precondition - in fact; it will return `NULL` if its invalid - so this `if` only short-circuits the return of `NULL`...and there are some callsites which are checking that the return value is null already...

I've removed it as it doesn't change behavior